src/chain/helpers/validatorStatus.ts

```diff
@@ -99,3 +101,59 @@ export function prepareValidatorForWithdrawal(state: BeaconState, index: Validat
   const validator = state.validatorRegistry[index];
   validator.withdrawalEpoch = getCurrentEpoch(state) + MIN_VALIDATOR_WITHDRAWAL_DELAY;
 }
+
+/**
+ * Return the committee assignment in the ``epoch`` for ``validator_index`` and ``registry_change``.
+ * ``assignment`` returned is a tuple of the following form:
+ * ``assignment[0]`` is the list of validators in the committee
+ * ``assignment[1]`` is the shard to which the committee is assigned
+ * ``assignment[2]`` is the slot at which the committee is assigned
+ * ``assignment[3]`` is a bool signalling if the validator is expected to propose
+ * a beacon block at the assigned slot.
+ * @param {BeaconState} state
+ * @param {Epoch} epoch
+ * @param {ValidatorIndex} validatorIndex
+ * @param {boolean} registryChange
+ * @returns {{validators: ValidatorIndex[]; shard: Shard; slot: number; isProposer: boolean}}
+ */
+export function getCommitteeAssignment(
```

Comment:
I think this function (and `isProposerAtSlot`) should go under `src/validator` somewhere.
This is something that the validator should calculate for themselves.

Reply:
I'm going to hold off on this until we architect a proper storage mechanism for the client